### PR TITLE
Add Unit Tests for Administration Profiles Section

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
Add unit tests and code climate fixes for the profiles section related files.

Coverage before:
===================== Coverage summary =====================
Statements   : 36.41% ( 1739/4776 )
Branches     : 10.99% ( 166/1511 )
Functions    : 30.82% ( 392/1272 )
Lines        : 36.41% ( 1739/4776 )
============================ ============================

Coverage after:
===================== Coverage summary =====================
Statements   : 42.33% ( 2024/4782 )
Branches     : 15.8% ( 238/1506 )
Functions    : 37.44% ( 477/1274 )
Lines        : 42.33% ( 2024/4782 )
============================= ============================

### CODE REVIEW

File: client/app/components/profiles/profile-editor.directive.js [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+      vm.descriptionValid = vm.editInfo.description && vm.editInfo.description.length > 0;
+      vm.providerTypeValid = vm.editInfo.providerType && vm.editInfo.providerType.length > 0;
+      vm.providerNameValid = vm.editInfo.providerName && vm.editInfo.providerName.length > 0;
+
+      vm.formValid = vm.nameValid && vm.descriptionValid && vm.providerTypeValid && vm.providerNameValid;
+    }
+
+    function getIdForName(list, name) {
+      if (list && list.length > 0) {
+        var found = list.find(function(nextObject) {
```
Comment (dtaylor113) [inline client/app/components/profiles/profile-editor.directive.js]: Is array.find() ES6?  Not sure this works in all browsers supported by MiQ.

File: client/app/components/profiles/profile-editor.directive.js [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+      var name;
+
+      if (angular.isDefined(namedObject)) {
+        name = namedObject.name;
+      }
+
+      return name;
+    }
+
+    function setInitialVars() {
```
Comment (dtaylor113) [inline client/app/components/profiles/profile-editor.directive.js]: Camel case?

**Comment by chriskacerguis** [comment]
@jeff-phillips-18 can you resolve the conflicts?

**Comment by jeff-phillips-18** [comment]
I am working on the find issue @dtaylor113 brought up. I will fix that then rebase and re-push.

**Comment by jeff-phillips-18** [comment]
Would like for these code climate issues to be allowed. We should create some utilities to put functions to perform these type of similar tasks.

**Comment by chriskacerguis** [comment]
@jeff-phillips-18 why not create the utility functions now?

**Comment by jeff-phillips-18** [comment]
@chriskacerguis OK, will do.

**Comment by chriskacerguis** [comment]
@jeff-phillips-18 ok, but that was more a genuine question than anything else.  I suspect that this exact situation will arise frequently, and I wonder what your thoughts are.

**Comment by jeff-phillips-18** [comment]
Based on the discussion in gitter, I thought we could try to address a bunch of the code climate issues all together. I can definitely kick off the effort in this PR by addressing the newly created issues. I'm game for tackling this either way.

**Comment by jeff-phillips-18** [comment]
Rethinking the utility since the function already exists in lodash. Sorry. Closing this PR and will resubmit.

**Pull request closed without merging**